fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlannerConfiguration.java

```diff
@@ -484,6 +492,18 @@ public Builder setPlanOtherAttemptWholeFilter(final boolean planOtherAttemptWhol
             return this;
         }
 
+        /**
+         * Set the maximum number of replanning-attempts during IN-to-JOIN transformations in the planner. This
+         * option only applies to {@link RecordQueryPlanner}.
+         * @param maxNumReplansForInToJoin the number of replanning attempts; defaults to {@code 0} for no
+         *        replanning attempts
+         * @return this builder
+         */
+        public Builder setMaxNumReplansForInToJoin(final int maxNumReplansForInToJoin) {
```

Comment:
Does this new planner configuration property allow for reverting back to the old behavior? I'm pretty sure the old behavior is worse, so I don't think we'd want to go back once this has rolled out, but arguably we might want it for plan stability during roll out.

Reply:
I think I don't have the confidence to provide a path where you get the exact same plan as before which as far as we know is squeal or word to the new behavior. I can provide yet another configuration that allows for enforcing to return the suboptimal plan after the initial planning knowing full well we create a worse plan. I am not a fan of that.